### PR TITLE
fixes some baton bugs

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

You can now turn on stun batons that have exactly enough charge left in them to stun someone.

Thrown stun baton impacts can now apply baton_effect() to most of the creatures that melee baton strikes can apply baton_effect() to.

## Why It's Good For The Game

Cattleprod strikes cost exactly 2000 points of charge to perform. A potato battery made from a 100 potency, capacitive cell production potato without the electrical activity trait grown by a prisoner will has a maximum charge of exactly 2000 points of charge. However, this prisoner cannot use that cell to power a single cattleprod strike (or to power a cattleprod so that it can be used in revival surgery to revive the prisoner's friend). This is very sad.

## Changelog
:cl: ATHATH
fix: You can now turn on stun batons that have exactly enough charge left in them to stun someone.
fix: Thrown stun batons can now debuff most of the types of creatures that melee stun baton strikes can.
/:cl:

### CODE REVIEW

File: code/game/objects/items/stunbaton.dm [hunk 1]
```diff
@@ -90,7 +90,7 @@
 /obj/item/melee/baton/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
 	..()
 	//Only mob/living types have stun handling
-	if(turned_on && prob(throw_stun_chance) && iscarbon(hit_atom))
+	if(turned_on && prob(throw_stun_chance) && isliving(hit_atom) && !iscyborg(hit_atom))
 		baton_effect(hit_atom)
```
Comment (LemonInTheDark) [inline code/game/objects/items/stunbaton.dm:94]: What does this actually effect?

Comment (ATH1909) [reply code/game/objects/items/stunbaton.dm:94]: batons don't just deal stamina damage and knock people down; they also cause them to stutter, confuse them, etc.

Comment (ATH1909) [reply code/game/objects/items/stunbaton.dm:94]: also, spiders care about the stamina damage

Comment (LemonInTheDark) [reply code/game/objects/items/stunbaton.dm:94]: Why not cyborgs?

Comment (AlinaStarkova) [reply code/game/objects/items/stunbaton.dm:94]: cyborgs dont take stamina damage no? they only stun.

Comment (ATH1909) [reply code/game/objects/items/stunbaton.dm:94]: > Why not cyborgs?

because the melee logic for stun batons excludes cyborgs but includes most other mobs

Comment (LemonInTheDark) [reply code/game/objects/items/stunbaton.dm:94]: But baton_effect has hardcoded behavior for them. What the fuck
image

Comment (AlinaStarkova) [reply code/game/objects/items/stunbaton.dm:94]: silicons arent carbon, the old code didnt apply to silicons, im not sure if that was intended or not but silicons dont recieve the same effects carbon mobs would when stunned, hence why im assuming it was intended for that to not call when hitting a silicon, there is probably a different line somewhere thats supposed to apply the silicon effects. though im not sure what exactly they are, do stun batons actually stun borgs still? im not actually sure what effects silicons are even supposed to get when hit with stun batons anymore

Comment (ATH1909) [reply code/game/objects/items/stunbaton.dm:94]: > But baton_effect has hardcoded behavior for them. What the fuck
> image

no, that's for if a borg is USING the baton, not for if it's being hit by one

Comment (LemonInTheDark) [reply code/game/objects/items/stunbaton.dm:94]: hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

**Pull request merged**